### ISSUE TITLE
Change the way for downloading DAQ packages' source under spack

### ISSUE THREAD

**Issue by dingp** [DUNE-DAQ/proto-spack#10]
For example, the `package.py` for `appfwk` will look like the following:

```python
class Appfwk(CMakePackage):
    """Home of daq_application and tools for writing DAQ modules"""

    homepage = "https://dune-daq-sw.readthedocs.io/en/latest/packages/appfwk/"
    git =      "https://github.com/DUNE-DAQ/appfwk.git"

    maintainers = ["jcfreeman2"]

    version("develop", branch="develop")
    version("N22-03-29", commit="HASH")
    version("2.3.4", commit="d2c1d1e03ff33dee9582f14c842a4c42584bb35d")
    version("2.3.3", commit="4d864d32af860f1c0e91e0597d0dcb09a63ce577")
    version("2.3.2", commit="3bc73fd95cba4d9b06d533e61d0554d30f62769f")
```

It currently looks ike:

```python
class Appfwk(CMakePackage):
    """Home of daq_application and tools for writing DAQ modules"""

    homepage = "https://dune-daq-sw.readthedocs.io/en/latest/packages/appfwk/"
    url =      "https://github.com/DUNE-DAQ/appfwk"

    maintainers = ["jcfreeman2"]

    version("develop", branch="develop", git="https://github.com/DUNE-DAQ/appfwk")
    version("2.3.4", sha256="c6a0e21374187e60e36f60758b1d05329b0ce9af0eaae5b633d285db485a64ca", extension="tar.gz", url="https://codeload.github.com/DUNE-DAQ/appfwk/legacy.tar.gz/dunedaq-v2.9.0")
    version('2.3.3', sha256='1683f6e72499c72e7ea2be9469b7d8dead06e5c65780d81ae82e4ce79dadd0fb', extension="tar.gz", url="https://codeload.github.com/DUNE-DAQ/appfwk/legacy.tar.gz/dunedaq-v2.8.2")
    version("2.3.2", sha256='348e629b9fe86b48690a6895cef46a0c6d67ab5a587514ce128e6dca156cc1cc', extension="tar.gz", url="https://codeload.github.com/DUNE-DAQ/appfwk/legacy.tar.gz/dunedaq-v2.8.0")
```

This change allows we simplify the entries into release YAML file, e.g. `dunedaq-v2.10.1.yaml` or `N22-03-29.yaml` for each package, as we only need to record the version (version name designated by spack) and commit hash for each package. And this is also the "trusted" method for downloading the sources.

**Comment by dingp** [comment]
I need to add a note about the proposed change. With the help of release YAML file and dedicated repo for each release, each `package.py` will be generated from a template and only contains two "version" lines, one for `develop`, and the other for the version used in the release. For example for `appfwk`, the template looks like:

```python
class Appfwk(CMakePackage):
    """Home of daq_application and tools for writing DAQ modules"""

    homepage = "https://dune-daq-sw.readthedocs.io/en/latest/packages/appfwk/"
    git =      "https://github.com/DUNE-DAQ/appfwk.git"

    maintainers = ["jcfreeman2"]

    version("develop", branch="develop")
    version("VERSION", commit="HASH")
```

`VERSION` and `HASH` will be substituted based on entries in each release YAML file.


I am wondering maybe we can even drop off the line of `version("develop", branch="develop")` too. @jcfreeman2 @alessandrothea  Please let me know your opinions on this.

**Comment by jcfreeman2** [comment]
This per-repo approach, where the commit hash rather than a tarball is used to safely download the code, looks good. I agree that if the thing that distinguishes each release of our packages (frozen or nightly) is the repo the package is in, we don't even need the develop branch build defined in the `package.py`.

**Comment by dingp** [comment]
Yeah, I prefer to remove `develop` too. That would reduce possible confusions since `spack list` will show the `develop` version of package to users if it sees that in `package.py`, but we don't plan to pre-install the `develop` version of any package.

**Comment by alessandrothea** [comment]
Thanks @dingp, looks good. I also agree that `develop` is not needed and should be removed.

**Comment by dingp** [comment]
This has been implemented in `daq-release`, feature branch `dingpf/spack-release`.

The following combination of commands will generate a release repo under `test-repo-outidr` for `dunedaq-v2.10.1`.
```bash
git clone http://github.com/DUNE-DAQ/daq-release -b dingpf/spack-release
cd daq-release/scripts/spack/
./make-release-repo.py -t ../../spack-repos/release-repo-template -i ../../configs/dunedaq-v2.10.1/dunedaq-v2.10.1.yaml -o ./test-repo-outdir
```

Calling the script in the following way will generate a release repo for the nightly release `N22-03-30` (as of today).
```
./make-release-repo.py -t ../../spack-repos/release-repo-template -i ../../configs/dunedaq-develop/dunedaq-develop.yaml -u -o ./test-nightly -r N22-03-30
```

**Comment by dingp** [comment]
Implemented in PR DUNE-DAQ/daq-release#128.

**Comment by dingp** [comment]
Done. PR https://github.com/DUNE-DAQ/daq-release/pull/128 has been merged.
